lib/Analysis/NoiseAnalysis/Params.h

```diff
@@ -0,0 +1,84 @@
+#ifndef INCLUDE_ANALYSIS_NOISEANALYSIS_PARAMS_H_
+#define INCLUDE_ANALYSIS_NOISEANALYSIS_PARAMS_H_
+
+#include <cmath>
+#include <vector>
+
+#include "llvm/include/llvm/Support/raw_ostream.h"  // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+class SchemeParam {
```

Comment:
I think these classes are fine, but I would prefer if we made two small tweaks:

1. Name them specific to the scheme they are operating on (if not the scheme + variant).
2. Put them in a new directory, maybe `lib/Parameters`.

The reason for (2) is that I suspect we will wrap these classes in IR attributes with custom parser/printers so we can attach them to the IR later. So having them in an isolated place allows us to avoid cyclic dependencies more easily.